draft-ietf-anima-brski-prm.md

```diff
@@ -1238,48 +1268,62 @@ In {{!RFC8995}}, the pledge generates PVR as CMS-signed JSON and PER as PKCS#10
-  The registrar MUST verify that the LDevID(ReAgt) certificate, corresponding to the signature, is still valid.
-  If the certificate is already expired, the registrar SHALL reject the request.
-  Validity of used signing certificates at the time of signing the agent-signed-data is necessary to avoid that a rogue Registrar-Agent generates agent-signed-data objects to onboard arbitrary pledges at a later point in time, see also {{sec_cons_reg-agt}}.
-  The registrar MUST fetch the EE (RegAgt) certificate, based on the provided SubjectKeyIdentifier (SKID) contained in the "kid" header parameter of the agent-signed-data, and perform this verification.
-  This requires, that the registrar has access to the EE (RegAgt) certificate data (including intermediate CA certificates if existent) based on the SKID.
-  Note, the registrar may have stored the EE (RegAgt) certificate if used during TLS establishment between Registrar-Agent and registrar or it may be provided via a repository.
+If the domain registrar is unable to return the Voucher, it SHOULD respond with an HTTP server error code.
+The following server error codes SHOULD be used:
 
-If the registrar is unable to validate the PVR, it SHOULD respond with a HTTP 4xx/5xx error code to the Registrar-Agent.
+* 500 Internal Server Error: if both Registrar-Agent request and MASA response are valid, but the registrar still failed to return the Voucher, e.g., due to missing configuration or a program failure
+* 502 Bad Gateway: if the registrar received an invalid response from the MASA
+* 503 Service Unavailable: if a simple retry of the Registrar-Agent request might lead to a successful response;
+  this error response SHOULD include the `Retry-After` response header field with an appropriate value
+* 504 Gateway Timeout: if the backend request to the MASA timed out
 
-The following 4xx client error codes SHOULD be used:
 
-* 403 Forbidden: if the registrar detected that one or more security related parameters are not valid or if the pledge-provided information could not be used with automated allowance.
+### Request Artifact: Pledge Voucher-Request (PVR)
 
-* 406 Not Acceptable: if the Content-Type indicated by the Accept header is unknown or unsupported.
+Identifical to the PVR artifact defined in {{pvr-artifact}}.
+The Registrar-Agent MUST NOT modify PVRs received from pledges.
 
-If the validation succeeds, the registrar performs pledge authorization according to {{Section 5.3 of !RFC8995}} followed by obtaining a voucher from the pledge's MASA according to {{Section 5.4 of !RFC8995}} with the modifications described below in {{rvr-proc}}.
 
-### Supply RVR to MASA (backend interaction) {#rvr-proc}
+### Backend Request Artifact: Registrar Voucher-Request (RVR) {#rvr-artifact}
 
 The registrar needs to convert the PVR to an RVR and supply it to the MASA.
 
```

Comment:
> If the MASA address/URI is learned from the IDevID MASA URI extension ({{Section 2.3 of !RFC8995}}), then the MASA on that URI MUST support the procedures defined in this document if the PVR used JSON-JWS encoding.
If the MASA is only configured on the registrar, then a registrar supporting BRKSI-PRM and other voucher encoding formats (such as those in {{!RFC8995}}) SHOULD support per-message-format MASA address/URI configuration for the same IDevID trust anchor."

It feels the important part is more that the PVR came to the registrar via Registrar-Agent, i.e., the pledge uses BRSKI-PRM and not normal BRSKI, rather than the JSW format.
Meaning the point seems to be if a pledge is manufactured with BRSKI-PRM support and has the MASA configured via IDevID, then the MASA must support BRSKI-PRM, i.e., its defaul format JWS Voucher.

Was that the intention of these statements?

Reply:
Yes, which actually sounds trivial as I read it. I'm not sure why we had a distinction between MASA URI in IDevID and MASA URI configured. As the MASA belongs to the manufacturer providing the pledge it should work. Hm, we are currently building up a MASA service with multi-tenant support. That means the MASA could also be used by other manufacturers. In that case, it is not that straight forward.

Reply:
Still, if at least one tenant produces devices with BRSKI-PRM, the overall MASA should (MUST) support its features including Agent-Proximity Assertion and the JWS artifact format.

I would like to simplify it to the statement that a MASA responsible for BRSKI-PRM capabable devices MUST support the BRSKI-PRM extensions. Maybe we list them then in Section 6 as central place for changes to be applied to components.

Reply:
I support the statement "that a MASA responsible for BRSKI-PRM capabable devices MUST support the BRSKI-PRM extensions."

Reply:
Done